lib/http-context.js

```diff
@@ -379,6 +406,11 @@ function toXML(input) {
  */
 
 HttpContext.prototype.done = function() {
+  if (this.method.stream) {
+    // the stream has already been setup
+    return;
+  }
+
```

Comment:
Is there any particular reason why you cannot handle streams in this `done` method? To me, it seems more logical to handle all different response arguments (like body stream) in a single place.